src/modules/launcher/Plugins/Microsoft.Plugin.Indexer/SearchHelper/OleDBSearch.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.Plugin.Indexer.Interface;
+using System;
+using System.Collections.Generic;
+using System.Data.OleDb;
+
+namespace Microsoft.Plugin.Indexer.SearchHelper
+{
+    public class OleDBSearch : ISearch
+    {       
+
+        public List<OleDBResult> Query(string connectionString, string sqlQuery)
+        {
+            List<OleDBResult> result = new List<OleDBResult>();
+
+            using (OleDbConnection conn = new OleDbConnection(connectionString))
+            {
+                // open the connection
+                conn.Open();
+
+                // now create an OleDB command object with the query we built above and the connection we just opened.
+                using (OleDbCommand command = new OleDbCommand(sqlQuery, conn))
+                {
+                    using (OleDbDataReader WDSResults = command.ExecuteReader())
+                    {
+                        if (WDSResults.HasRows)
+                        {
+                            while (WDSResults.Read())
+                            {
+                                List<Object> fieldData = new List<object>();
+                                for (int i = 0; i < WDSResults.FieldCount; i++)
+                                    fieldData.Add(WDSResults.GetValue(i));
```

Comment:
I don't think we need to retrieve more data than what is needed. Probably we should do a 
min (2, WDSResults.FieldCount )?

Reply:
We aren't retrieving more that here. `WDSResults.FieldCount` will be 3 based on number of columns that we have requested from the indexer database i.e "System.ItemUrl, System.FileName, System.FileAttributes"

Reply:
Thanks for the clarification. I have a tangential question about the FieldCount value. What would happen when some column does not exist for a file (say it does not have attributes), then the FieldCount value might reduce to less than 3. It might lead to a possibility where we might be missing some column which might result in an exception when we try to access the fieldData value which has not been populated.

Reply:
`FieldCount` is a calculated value based on the number of columns in a record returned by the database. 

Now, Based on your question there could be two scenarios : 
1. If the query contains a column that doesn't exist, `OleDbDataReader WDSResults = command.ExecuteReader()` fails with following error : `<System.Data.OleDb.OleDbException (0x80040E55): Column does not exist.`
The subsequent error handling is been done in `Main.js` for indexer plugin. 
2. If there is no data for a file in a column, it will be set to `DBNull.value`. Since `OleDBSearch` is class with the sole purpose of getting back results for a query, we are handling this case in `WindowsSearchAPI` class.